unwrap.go

```diff
@@ -9,18 +9,36 @@ import "runtime"
 //
 // You can use this for structured access to trace information.
 func UnwrapFrame(err error) (frame runtime.Frame, inner error, ok bool) { //nolint:revive // error is intentionally middle return
-	e, ok := err.(*errTrace)
+	e, ok := err.(interface{ TracePC() uintptr })
 	if !ok {
 		return runtime.Frame{}, err, false
 	}
 
-	frames := runtime.CallersFrames([]uintptr{e.pc})
+	wrapErr := unwrapOnce(err)
+	frames := runtime.CallersFrames([]uintptr{e.TracePC()})
 	f, _ := frames.Next()
 	if f == (runtime.Frame{}) {
 		// Unlikely, but if PC didn't yield a frame,
 		// just return the inner error.
-		return runtime.Frame{}, e.err, false
+		return runtime.Frame{}, wrapErr, false
 	}
 
-	return f, e.err, true
+	return f, wrapErr, true
+}
+
+// unwrapOnce accesses the direct cause of the error if any, otherwise
+// returns nil.
+//
+// It supports both errors implementing causer (`Cause()` method, from
+// github.com/pkg/errors) and `Wrapper` (`Unwrap()` method, from the
+// Go 2 error proposal).
```

Comment:
we only support unwrapping using `Unwrap()` in `tree.go` currently,
https://github.com/bracesdev/errtrace/blob/main/tree.go#L51-L57

My preference is to only support `Unwrap() error` for consistency, and separately we can consider adding support for `Cause() error` (but add it throughout).

Reply:
Ok, I removed that case for now.